File: Museo Java/ObraDeArte.java
import java.time.LocalDate;

public abstract class ObraDeArte {

    public static final String[] ESTADOS_VALIDOS = {"expuesta", "en_restauracion", "cedida"};

    protected int idObra;
    protected String titulo;
    protected String autor;
    protected String periodo;
    protected double valoracion;
    protected LocalDate fechaCreacion;
    protected LocalDate fechaEntrada;
    protected String estado;

    public ObraDeArte(int idObra, String titulo, String autor, String periodo,
                      double valoracion, LocalDate fechaCreacion,
                      LocalDate fechaEntrada, String estado) {
        this.idObra = idObra;
        this.titulo = titulo;
        this.autor = autor;
        this.periodo = periodo;
        this.valoracion = valoracion;
        this.fechaCreacion = fechaCreacion;
        this.fechaEntrada = fechaEntrada;
        this.estado = estado;
    }

    public ObraDeArte(int idObra, String titulo, String autor, String periodo,
                      double valoracion, LocalDate fechaCreacion, LocalDate fechaEntrada) {
        this(idObra, titulo, autor, periodo, valoracion, fechaCreacion, fechaEntrada, "expuesta");
    }

    public int getIdObra() { return idObra; }
    public String getTitulo() { return titulo; }
    public String getAutor() { return autor; }
    public String getPeriodo() { return periodo; }
    public double getValoracion() { return valoracion; }
    public LocalDate getFechaCreacion() { return fechaCreacion; }
    public LocalDate getFechaEntrada() { return fechaEntrada; }
    public String getEstado() { return estado; }

    public void setEstado(String nuevoEstado) {
        boolean valido = false;
        for (String e : ESTADOS_VALIDOS) {
            if (e.equals(nuevoEstado)) { valido = true; break; }
        }
        if (!valido) {
            throw new IllegalArgumentException(
                "Estado inválido: '" + nuevoEstado + "'. Debe ser uno de: expuesta, en_restauracion, cedida."
            );
        }
        this.estado = nuevoEstado;
    }

    public abstract String consultarInfo();

    @Override
    public String toString() {
        return getClass().getSimpleName() + "(id=" + idObra +
               ", titulo='" + titulo + "', estado='" + estado + "')";
    }
}

File: MuseoPOO/obra_de_arte.py
"""Módulo que define la clase abstracta ObraDeArte."""

from abc import ABC, abstractmethod
from datetime import date


class ObraDeArte(ABC):
    """Clase abstracta que representa una obra de arte."""

    ESTADOS_VALIDOS = ("expuesta", "en_restauracion", "cedida")

    def __init__(
        self,
        id_obra: int,
        titulo: str,
        autor: str,
        periodo: str,
        valoracion: float,
        fecha_creacion: date,
        fecha_entrada: date,
        estado: str = "expuesta",
    ) -> None:
        self._id_obra = id_obra
        self._titulo = titulo
        self._autor = autor
        self._periodo = periodo
        self._valoracion = valoracion
        self._fecha_creacion = fecha_creacion
        self._fecha_entrada = fecha_entrada
        self._estado = estado

    def get_id_obra(self) -> int:
        return self._id_obra

    def get_titulo(self) -> str:
        return self._titulo

    def get_autor(self) -> str:
        return self._autor

    def get_periodo(self) -> str:
        return self._periodo

    def get_valoracion(self) -> float:
        return self._valoracion

    def get_fecha_creacion(self) -> date:
        return self._fecha_creacion

    def get_fecha_entrada(self) -> date:
        return self._fecha_entrada

    def get_estado(self) -> str:
        return self._estado

    def set_estado(self, nuevo_estado: str) -> None:
        if nuevo_estado not in self.ESTADOS_VALIDOS:
            raise ValueError(
                f"Estado inválido: '{nuevo_estado}'. "
                f"Debe ser uno de {self.ESTADOS_VALIDOS}."
            )
        self._estado = nuevo_estado

    @abstractmethod
    def consultar_info(self) -> str:
        """Retorna información detallada de la obra de arte."""

    def __str__(self) -> str:
        return (
            f"{self.__class__.__name__}(id={self._id_obra}, "
            f"titulo='{self._titulo}', estado='{self._estado}')"
        )
